Decred Wallet/Custom Views/FloatingLabelTextInput.swift

```diff
@@ -0,0 +1,166 @@
+//
+//  FloatingLabelTextInput.swift
+//  Decred Wallet
+//
+// Copyright (c) 2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import UIKit
+
+@IBDesignable
+class FloatingLabelTextInput: UITextField {
+    let border = CALayer()
+    let borderWidth: CGFloat = 2
+    var activeBorderColor: UIColor = UIColor.appColors.decredBlue
+    let inactiveBorderColor: UIColor = UIColor.appColors.lightGray
+    
+    let cornerRadius: CGFloat = 4
+    
+    var floatingLabel = PaddedLabel()
+    var floatingLabelBottomConstraint:NSLayoutConstraint?
+    let floatingLabelBottomPaddingNormal:CGFloat = 33
+    let floatingLabelBottomPaddingActive:CGFloat = 8
+    let floatingLabelColorNormal:UIColor = UIColor.appColors.bluishGray
+    var floatingLabelColorActive:UIColor = UIColor.appColors.decredBlue
+    
+    var button = UIButton(type: .custom)
+    
+    var isActive:Bool = false
+    let padding = UIEdgeInsets(top: 0, left: 16, bottom: 0, right: 0)
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.initView()
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.initView()
+    }
+    
+    private func initView() {
+        border.cornerRadius = self.cornerRadius
+        border.borderColor = self.inactiveBorderColor.cgColor
+        border.borderWidth = self.borderWidth
+        self.layer.addSublayer(border)
+        
+        self.initFloatedLabel()
+        self.addTarget(self, action: #selector(self.editingBegan), for: .editingDidBegin)
+        self.addTarget(self, action: #selector(self.editingEnded), for: .editingDidEnd)
+    }
+    
+    private func initFloatedLabel() {
+        self.floatingLabel.backgroundColor = UIColor.white
+        self.floatingLabel.textColor = self.floatingLabelColorNormal
+        self.floatingLabel.font = UIFont(name: self.font!.familyName, size: 16)
+        self.floatingLabel.clipsToBounds = true
+        self.floatingLabel.translatesAutoresizingMaskIntoConstraints = false
+        self.floatingLabel.sizeToFit()
+        self.floatingLabel.leftPadding = 4
+        self.floatingLabel.rightPadding = 4
+        self.floatingLabel.text = self.placeholder
+        self.placeholder = ""
+        self.addSubview(self.floatingLabel)
+        self.floatingLabelBottomConstraint = self.floatingLabel.bottomAnchor.constraint(equalTo: self.topAnchor, constant: self.floatingLabelBottomPaddingNormal)
+        NSLayoutConstraint.activate([
+            self.floatingLabel.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: 8),
+            self.floatingLabelBottomConstraint!
+        ])
+    }
+    
+    func setPlaceHolder(text:String) {
+        self.floatingLabel.text = text
+    }
```

Comment:
Instead of this, I'd recommend overriding the `placeholder` property and updating the label text on `didSet`.
```
override var placeholder: String? {
    didSet {
        self.floatingLabel.text = placeholder
        placeholder = ""
    }
}
```

This would mean that you cannot set `placeholder = ""` from anywhere else as this will also clear the text in the label.